### PR TITLE
Resumable Upload: Add details about Transfer-Encoding and Content-Encoding

### LINKED ISSUE #2679: Speak about considerations of byte counting and transfer-encoding
Ted Hardie asked:

> How does the upload offset calculation interact with transfer encoding/compression? I'm reporting how many bytes came through the network, is that right?

We probably would benefit from some consideration text to mention how HTTP/1.1 transfer-encoding might be a footgun if the server counted those bytes naiively.

### PR DESCRIPTION
**Pull request by Acconut**
With this PR, I attempt to clarify the role of content and transfer encoding in resumable upload that is:
- the bytes for offsets are counted on a representation, where Content-Encoding might be applied, but not Transfer-Encoding because that is a property of the transfer and not the representation itself.
- an upload retain the same content coding throughout its lifecycle, meaning that requests to resume the upload should target the same representation with the same content coding. Otherwise, data might end up invalid on the server if the content coding is switched during the upload. 

Closes #2679 and #2674.

### CODE REVIEW

File: draft-ietf-httpbis-resumable-upload.md [hunk 1]
```diff
@@ -266,7 +268,7 @@ Upload-Offset: 25
...
+File metadata can affect how servers might act on the uploaded file. Clients can send representation metadata (see {{Section 8.3 of HTTP}}) in the request that starts an upload. Servers MAY interpret this metadata or MAY ignore it. The `Content-Type` header field ({{Section 8.3 of HTTP}}) can be used to indicate the MIME type of the file. The content coding of the representation is specified using the `Content-Encoding` header field and is retained throughout the entire upload. When resuming an interrupted upload, the same content coding is used for appending to the upload, producing a representation of the upload resource with one consistent content coding. The `Content-Disposition` header field ({{!RFC6266}}) can be used to transmit a filename; if included, the parameters SHOULD be either `filename`, `filename*` or `boundary`.
```
Comment (smatsson) [inline draft-ietf-httpbis-resumable-upload.md:271]: Minor nit, but is "coding" and "encoding" the same thing? Google doesn't give a straight answer and English is not my first language.

File: draft-ietf-httpbis-resumable-upload.md [hunk 2]
```diff
@@ -423,6 +425,10 @@ The `Upload-Complete` header field MUST only be used if support by the resource
...
 
+# Transfer and Content Codings
+
+A message might have a content coding, indicated by the `Content-Encoding` header field, and/or a transfer coding, indicated by the `Transfer-Encoding` header field ({{Section 6.1 of RFC9112}}), applied, which modify the representation of uploaded data in a message. For correct interoperability, the client and server must share the same logic when counting bytes for the upload offset. From the client's perspective, the offset is counted after content coding but before transfer coding is applied. From the server's perspective, the offset is counted after the content's transfer coding is reversed but before the content coding is reversed.
```
Comment (smatsson) [inline draft-ietf-httpbis-resumable-upload.md:430]: Good addition! :) Same with coding vs encoding here

File: draft-ietf-httpbis-resumable-upload.md (hunk 1 above)
Comment (Acconut) [reply draft-ietf-httpbis-resumable-upload.md:271]: I took the language from [RFC 9110](https://httpwg.org/specs/rfc9110.html#field.content-encoding):
> The "Content-Encoding" header field indicates what content codings have been applied to the representation

So Content-Encoding is the header field name and its value identifies the used content codings, if I understood it correctly.

File: draft-ietf-httpbis-resumable-upload.md (hunk 2 above)
Comment (Acconut) [reply draft-ietf-httpbis-resumable-upload.md:430]: Same as above, but from RFC 9112: https://datatracker.ietf.org/doc/html/rfc9112#name-transfer-codings

File: draft-ietf-httpbis-resumable-upload.md (hunk 1 above)
Comment (smatsson) [reply draft-ietf-httpbis-resumable-upload.md:271]: Thanks for looking it up! Seems reasonable. Ignore my minor nit :)

**Review by guoye-zhang — APPROVED** [review]

**Pull request merged**